File: Go/ExampleTests/common/match_string.go
package common

type MatchString struct {
	MatchedAddress string
	AddressComponents AddressComponentsString
}

func NewMatchStringFromSlice(v []string) MatchString {
	s := MatchString{}
	if len(v) > 0 { s.MatchedAddress = v[0] }
	if len(v) > 1 { s.AddressComponents = NewAddressComponentsStringFromText(v[1]) }
	return s
}

// NewMatchStringFromText builds from the text form, e.g. Money as "25 USD".
func NewMatchStringFromText(text string) MatchString {
	parts := RequireTokens(text, 2, "Match")
	return MatchString{
		MatchedAddress: parts[0],
		AddressComponents: NewAddressComponentsStringFromText(parts[1]),
	}
}

func (s MatchString) String() string {
	return Token(s.MatchedAddress) + " " + Nested(s.AddressComponents.String())
}

func (s MatchString) Equals(o MatchString) bool {
	return DNCEqual(s.MatchedAddress, o.MatchedAddress) &&
		s.AddressComponents.Equals(o.AddressComponents)
}

func EqualMatchStringSlices(a, b []MatchString) bool {
	if len(a) != len(b) { return false }
	for i := range a {
		if !a[i].Equals(b[i]) { return false }
	}
	return true
}

File: Rust/ExampleTests/common/match__string.rs
#![allow(dead_code, unused_imports, unused_variables)]

use super::addresscomponents_string::AddressComponentsString;
#[derive(Debug, Clone, Default)]
pub struct MatchString {
    pub matchedaddress: String,
    pub addresscomponents: AddressComponentsString,
}

impl MatchString {
    pub fn from_vec(v: &[&str]) -> Self {
        Self {
            matchedaddress: v.get(0).copied().unwrap_or("").to_string(),
            addresscomponents: Default::default(),
        }
    }

    /// Builds from the text form, e.g. Money as "25 USD".
    pub fn from_text(text: &str) -> Self {
        let parts = crate::common::tokens::require(text, 2, "Match");
        Self {
            matchedaddress: parts[0].clone(),
            addresscomponents: AddressComponentsString::from_text(&parts[1]),
        }
    }
}

impl std::fmt::Display for MatchString {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        write!(f, "{}", [crate::common::tokens::token(&self.matchedaddress), crate::common::tokens::nested(&self.addresscomponents.to_string())].join(" "))
    }
}

impl PartialEq for MatchString {
    fn eq(&self, other: &Self) -> bool {
        crate::common::dnc_equal(&self.matchedaddress, &other.matchedaddress)
            && self.addresscomponents == other.addresscomponents
    }
}
